### PR TITLE
CORDA-1912: Add documentation on the options for deploying nodes

### PR DESCRIPTION
**Pull request by fowlerrr**
Added some clarification on the available options for Cordform tasks.

### CODE REVIEW

File: docs/source/generating-a-node.rst [hunk 1]
```diff
@@ -106,6 +106,22 @@ Additionally, all three nodes will include any CorDapps defined in the project's
...
+* ``notary`` for notaries only (see :ref:`notary <corda_configuration_file_notary>`)
+* ``p2pAddress`` the address to open for the p2p connection (see :ref:`p2pAddress <corda_configuration_file_p2pAddress>`)
+* ``p2pPort`` the port to open for the p2p connection (see :ref:`p2pAddress <corda_configuration_file_p2pAddress>`)
+* ``devMode`` when true enables development mode (see :ref:`devMode <corda_configuration_file_dev_mode>`)
+* ``webPort`` (see :ref:`Specifying a custom webserver<specify-custom-webserver>`)
+* ``webAddress`` (see :ref:`Specifying a custom webserver<specify-custom-webserver>`)
+* ``rpcSettings`` (see :ref:`rpcSettings <corda_configuration_file_rpc_settings>`)
+* ``rpcUsers`` (see :ref:`rpcUsers <corda_configuration_file_rpc_users>`)
+* ``configFile`` for extending configuration of nodes (see below)
+* ``https`` when true enables HTTPS communication
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: For what? Is this just for internode communication? Or does this apply to RPC as well?

File: docs/source/generating-a-node.rst [hunk 2]
```diff
@@ -106,6 +106,22 @@ Additionally, all three nodes will include any CorDapps defined in the project's
...
 template CorDapp, for example, would automatically build and add the template CorDapp to each node.
 
+The configuration values available in ``deployNodes`` task are as follows:
+
+* ``name`` the name of the Corda node (see :ref:`myLegalName <corda_configuration_file_myLegalName>`)
+* ``notary`` for notaries only (see :ref:`notary <corda_configuration_file_notary>`)
+* ``p2pAddress`` the address to open for the p2p connection (see :ref:`p2pAddress <corda_configuration_file_p2pAddress>`)
+* ``p2pPort`` the port to open for the p2p connection (see :ref:`p2pAddress <corda_configuration_file_p2pAddress>`)
+* ``devMode`` when true enables development mode (see :ref:`devMode <corda_configuration_file_dev_mode>`)
+* ``webPort`` (see :ref:`Specifying a custom webserver<specify-custom-webserver>`)
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: For this and the ones below, would be good to give a quick description for ease

File: docs/source/generating-a-node.rst [hunk 3]
```diff
@@ -106,6 +106,22 @@ Additionally, all three nodes will include any CorDapps defined in the project's
...
+* ``webAddress`` (see :ref:`Specifying a custom webserver<specify-custom-webserver>`)
+* ``rpcSettings`` (see :ref:`rpcSettings <corda_configuration_file_rpc_settings>`)
+* ``rpcUsers`` (see :ref:`rpcUsers <corda_configuration_file_rpc_users>`)
+* ``configFile`` for extending configuration of nodes (see below)
+* ``https`` when true enables HTTPS communication
+* ``h2Settings`` (see :ref:`h2Settings <corda_configuration_file_h2_settings>`)
+* ``sshdPort`` Specifies the port for sshd communication (see :ref:`sshd <corda_configuration_file_sshd>`)
+
 You can extend ``deployNodes`` to generate additional nodes.
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: Hey, while you're here, would you be able to update this line? I assume that they're saying you can have as many node blocks inside deployNodes as you want, with each one creating a new node.

File: docs/source/generating-a-node.rst [hunk 4]
```diff
@@ -106,6 +106,22 @@ Additionally, all three nodes will include any CorDapps defined in the project's
 CorDapps are not listed in each node's ``cordapps`` entry. This means that running the ``deployNodes`` task from the
 template CorDapp, for example, would automatically build and add the template CorDapp to each node.
 
+The configuration values available in ``deployNodes`` task are as follows:
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: Either here or in the linked sections, we need to give example values. What is the standard format for specifying the `p2pAddress`? Should true/false be upper or lower case? etc.

File: docs/source/generating-a-node.rst [hunk 5]
```diff
@@ -106,6 +106,22 @@ Additionally, all three nodes will include any CorDapps defined in the project's
...
+
+* ``name`` the name of the Corda node (see :ref:`myLegalName <corda_configuration_file_myLegalName>`)
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: I'd say "the legal identity name" here.

File: docs/source/generating-a-node.rst [hunk 6]
```diff
@@ -106,6 +106,22 @@ Additionally, all three nodes will include any CorDapps defined in the project's
...
+* ``notary`` for notaries only (see :ref:`notary <corda_configuration_file_notary>`)
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: What for the notaries only? We should specify.

File: docs/source/generating-a-node.rst (hunk 4 above)
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: Can we also specify which ones are optional, and which are required?

File: docs/source/generating-a-node.rst [hunk 7]
```diff
@@ -106,6 +106,22 @@ Additionally, all three nodes will include any CorDapps defined in the project's
...
+* ``p2pAddress`` the address to open for the p2p connection (see :ref:`p2pAddress <corda_configuration_file_p2pAddress>`)
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: This isn't very clear to me. What is the p2p connection? I'd suggest something like "the address the node advertises for inbound connections from other nodes" or something

File: docs/source/generating-a-node.rst [hunk 8]
```diff
@@ -106,6 +106,22 @@ Additionally, all three nodes will include any CorDapps defined in the project's
...
+* ``p2pPort`` the port to open for the p2p connection (see :ref:`p2pAddress <corda_configuration_file_p2pAddress>`)
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: As above

File: docs/source/generating-a-node.rst (hunk 1 above)
Comment (fowlerrr) [reply docs/source/generating-a-node.rst]: Best I can tell it's for the webserver the node sets up.  I've added a couple of words to (hopefully) clarify that.

File: docs/source/generating-a-node.rst (hunk 3 above)
Comment (fowlerrr) [reply docs/source/generating-a-node.rst]: Done

File: docs/source/generating-a-node.rst (hunk 5 above)
Comment (fowlerrr) [reply docs/source/generating-a-node.rst]: Done

File: docs/source/generating-a-node.rst (hunk 4 above)
Comment (fowlerrr) [reply docs/source/generating-a-node.rst]: Turns out not many are required.  Let me know about the formatting of them.

File: docs/source/generating-a-node.rst (hunk 7 above)
Comment (fowlerrr) [reply docs/source/generating-a-node.rst]: Done as suggested

File: docs/source/generating-a-node.rst (hunk 6 above)
Comment (fowlerrr) [reply docs/source/generating-a-node.rst]: Yeah that was vague.  Hopefully it's clearer now

File: docs/source/generating-a-node.rst [hunk 9]
```diff
@@ -108,56 +108,118 @@ template CorDapp, for example, would automatically build and add the template Co
...
+    ::
 
-  * For extending configuration of nodes.
+        configFile = "samples/trader-demo/src/main/resources/node-b.conf"
 
-* ``https``
+* ``https`` <boolean>
 
-  * When true enables HTTPS communication from the node webserver.
+  * When true enables HTTPS communication from the node webserver. e.g.
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: Doubling-up of e.g.

File: docs/source/generating-a-node.rst [hunk 10]
```diff
@@ -108,56 +108,118 @@ template CorDapp, for example, would automatically build and add the template Co
 
 The configuration values available in ``deployNodes`` task are as follows:
 
-* ``name`` <**required**> (see :ref:`myLegalName <corda_configuration_file_myLegalName>`)
+Required configuration
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: If you check the generated page, the `Required configuration` and `Optional configuration` headers are at a higher level than the other three in the `The Cordform task` section, which doesn't seem right.

**Pull request merged**